lambdaclass/afk_gacha_game
# [GH-438] Implement dungeon settlement level up. Fix currency assets

> [!CAUTION]
> To be merged with https://github.com/lambdaclass/mirra_backend/pull/656

Closes #438 

## Motivation

We need to level up the Dungeon Settlement, being able to see the costs in the screen.

## Summary of changes

- Implement a DungeonSettlementManager that will handle most of the gameobjects that display information or perform an action in the Dungeon Settlement scene.
- Implement the functionality of the level up button.
- Update gateway.proto with the necessary changes 

## How has this been tested?

1. Go to the DungeonSettlement scene. 
2. You should see 50 blueprints available to spend in the header, and a cost of 100 gold + 50 blueprints to level up. Your settlement's level should be 1. 
3. Click/tap on the level up button. The level should increment in 1, the blueprints amount in the header should decrease to 0 and the supplies' AFK Rate should not be 0 anymore.
4. Try to level up again. You should get a pop up informing that currency is not enough.

## Checklist
